src/analysis/util.cc

```diff
@@ -25,131 +25,98 @@ const std::vector<Piece> kPieceSet{
 };
 }  // namespace
 
-std::vector<Piece> getRiichiDiscard(std::vector<Piece> hand) {
-  if (hand.empty()) {
+std::vector<Piece> completeSet(const Piece a, const Piece b) {
+  if (a.getSuit() != b.getSuit()) {
     return {};
   }
-  std::array<int8_t, Piece::kPiecesize> counts = {};
-  std::array<bool, Piece::kPiecesize> removedbefore = {};
-  std::vector<Piece> remove_me;
-  for (const auto& p : hand) {
-    counts.at(p.toUint8_t())++;
+  const int piece_diff = static_cast<int>(a.getPieceNum()) - b.getPieceNum();
+  if (std::abs(piece_diff) > 2) {
+    return {};
   }
-  for (int i = 0; i < 9; i++) {
-    const Piece removed = hand.front();
-    hand.erase(hand.begin());
-    if (removedbefore.at(removed.toUint8_t())) {
-      hand.push_back(removed);
-      continue;
-    }
-    removedbefore.at(removed.toUint8_t()) = true;
-    for (const auto& p : kPieceSet) {
-      if (counts.at(p.toUint8_t()) == 4 || p == removed) {
-        continue;
-      }
-      hand.push_back(p);
-      auto root = breakdownHand(hand);
-      if (root->IsComplete()) {
-        remove_me.push_back(removed);
-      }
-
-      hand.pop_back();
-    }
-    hand.push_back(removed);
+  if (std::abs(piece_diff) != 1) {
+    return {b + (piece_diff >> 1)};
```

Comment:
Can you indicate what the bit shifts purpose is here with a comment? I know the usage is consistent but would be nice for quick understanding. Im assuming suit?